File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
import sys
input = sys.stdin.readline

n = int(input())
arr = sorted(list(map(int, input().split())))
x = int(input())

left = 0
right = n - 1

cnt = 0
while left < right:
  if arr[left] + arr[right] > x:
    right -= 1
  elif arr[left] + arr[right] < x:
    left += 1
  else:
    cnt += 1
    left += 1
    right -= 1
print(cnt)

File: 백준/Silver/3273. 두 수의 합/두 수의 합.js
const input = require('fs').readFileSync(process.platform == 'win32' ? "./input.txt" : '/dev/stdin').toString().trim().split("\n")

const length = Number(input[0]);
const arr = input[1].split(" ").map(Number).sort((a,b) => a-b); // 오름차순 정렬렬
const x = Number(input[2]);

const solution = () => {
    let count = 0;
    let left = 0;
    let right = length - 1;

    while (left < right) {
        let sum = arr[left] + arr[right];
        
        if (sum == x) {
            count++;
            left++;
        } else if (sum < x) {
            left++;
        } else {
            right--;
        }
    }
    console.log(count);
}

solution();